### PR TITLE
MLIBZ-1802: Logs

### PR DESCRIPTION
**Pull request by thomasconner**
#### Description
This PR adds logging to the SDK.

#### Changes
- Log outgoing requests and incoming responses.
- Log active user activity.
- Log queries.

### CODE REVIEW

File: src/request/src/request.js [hunk 1]
```diff
@@ -177,6 +180,7 @@ export default class Request {
 
   toPlainObject() {
     return {
+      id: this.id,
```
Comment (tejasranade) [inline src/request/src/request.js]: @thomasconner Is there any other value to the `id` outside of logging it here? Do we tie the request ID, for example, through the pipeline that the request goes through?

File: src/utils/src/log.js [hunk 2]
```diff
@@ -18,13 +22,9 @@ Log.methodFactory = function methodFactory(methodName, logLevel, loggerName) {
     }
   };
 };
+log.setLevel(log.getLevel());
```
Comment (tejasranade) [inline src/utils/src/log.js]: @thomasconner this looks odd. This should be a No-Op if getters and setters do what they're supposed to :)

File: package.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
     "hellojs": "1.14.1",
     "local-storage": "1.4.2",
     "lodash": "4.17.4",
-    "loglevel": "1.4.1",
+    "loglevel": "github:thomasconner/loglevel",
```
Comment (tejasranade) [inline package.json]: @thomasconner what were the changes you made to the package? Do you plan to keep your fork updated periodically from the master?

File: src/query.js [hunk 4]
```diff
@@ -763,6 +768,8 @@ export default class Query {
     const json = this.toPlainObject();
     data = sift(json.filter, data);
 
+    Log.debug('After applying query filter to data', data);
```
Comment (tejasranade) [inline src/query.js]: @thomasconner Have you considered that this data could be quite large and flood the log? Would it make sense to truncate or just print the count instead, for example?

File: src/request/src/request.js (hunk 1 above)
Comment (thomasconner) [inline src/request/src/request.js]: This `id` is generated by the SDK and used to match up log statements for requests. Multiple requests might be logged at the same time and without some kind of identifier it would be hard to match log statements that are related. We need to communicate that this is not related to the `X-Kinvey-Request-Id` header by adding the information to the troubleshooting guide.

Comment (thomasconner) [inline src/request/src/request.js]: This is the related [JIRA ticket](https://kinvey.atlassian.net/browse/MLIBZ-2025).

File: package.json (hunk 3 above)
Comment (thomasconner) [inline package.json]: I submitted a [PR](https://github.com/pimterry/loglevel/pull/104) that was merged into the loglevel master branch. We can use the latest release of loglevel instead of a personally maintained version.
